design/handle-backup-of-volumes-by-resources-filters.md

```diff
@@ -0,0 +1,223 @@
+# Handle backup of volumes by resources filters
+ 
+## Abstract
+Currently, Velero doesn't have one flexible way to filter volumes.
+ 
+If users want to skip backup of volumes or only backup some volumes in different namespaces in batch, currently they need to use the opt-in and opt-out approach one by one, or use label-selector but if it has big different labels on each different related pods, which is cumbersome when they have lots of volumes to handle with. it would be convenient if Velero could provide one way to filter the backup of volumes just by `some specific volumes attributes`.
+ 
+Also, currently, it's not accurate enough if the users want to select a specific volume to do a backup or skip by without patching labels or annotations to the pods. It would be useful if users could accurately select target volume by `one specific resource selector`. Users could accurately select the volume to backup or skip in their own console when using velero for secondary development.
+ 
+## Background
+As of Today, Velero has lots of filters to handle (backup or skip backup) resources including resources filters like `IncludedNamespaces, ExcludedNamespaces`, label selectors like `LabelSelector, OrLabelSelectors`, annotation like `backup.velero.io/must-include-additional-items` etc. But it's not enough flexible to handle volumes, we need one generic way to filter volumes.
+ 
+## Goals
+- Introducing one flexible way to filter volumes.
+ 
+## Non Goals
+- We only handle volumes for backup and do not support restore.
+- Currently, only handle volumes, does not support other resources.
+- Only environment-unrelated and platform-independent general volumes attributes are supported, do not support volumes attributes related to a specific environment.
+ 
+## Use-cases/Scenarios
+### A. Skip backup volumes by some attributes
+Users want to skip PV with the requirements:
+- option to skip all PV data
+- option to skip specified PV type (RBD, NFS)
+- option to skip specified PV size
+- option to skip specified storage-class
+- option to skip folders
+ 
+### B. Accurately select target volume to backup
+Some volumes are only used for logging while others volumes are for DBs, and only need to backup DBs data. users need `one specific resource seletor` to accurately select target volumes to backup.
+ 
+## High-Level Design
+Add a new flag `from-file` when executing `velero backup create`, which imports the defined resources filters in one JSON file. the JSON file including all defined filter rules for the current backup.
+ 
+When Velero handles volumes backup should respect the filter rules defined in the imported JSON file.
+ 
+## Detailed Design
+The resources filters rules should contain both `include` and `exclude` rules.
```

Comment:
Let's focus on PVs, not other resources.

If we define multiple filters in one config, what's the relationship between the filters?  My initial thought is that it should be OR, such that each `include` and `exclude` rule can be simple and it may be possible to satisfy the scenarios like for PVs that meet criteria in filter A we take a snapshot, for PVs that meet criteria in filter B we use kopia?

I am open for discussion, admitting if we wanna support different actions for different PVs, things can be tricky.

Reply:
I'm not sure `meet criteria` is easy to confirm with a specific user environment

Reply:
Like you mentioned in 
https://github.com/vmware-tanzu/velero/pull/5773/files#r1090315744
Worst case the user can choose to use GVRN...